src/game_api/spawn_api.hpp

```diff
@@ -49,7 +49,7 @@ void pop_spawn_type_flags(SPAWN_TYPE flags);
 
 void init_spawn_hooks();
 
-void spawn_player(int8_t player_slot, float x, float y);
+int32_t spawn_player(int8_t player_slot, std::optional<float> x = std::nullopt, std::optional<float> y = std::nullopt, std::optional<LAYER> layer = std::nullopt);
```

Comment:
autodoc will probably not like defaults, and kind of no reason to have them as sol doesn't care and you wrote separate UI version